src/main/java/org/prebid/server/auction/CpmRange.java

```diff
@@ -54,7 +54,7 @@ private static ExtGranularityRange findRangeFor(BigDecimal cpm, List<ExtGranular
             }
             min = range.getMax();
         }
-        throw new IllegalArgumentException("Cpm is negative or price granularity ranges are not valid.");
+        return null;
```

Comment:
If `null` happens to return from this method then `fromCpmAsNumber` will experience NPE. This wouldn't happen if piece of `fromCpmAsNumber` calling this method was reverted back to using streams and optional.